c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -1283,6 +1279,26 @@ class PostgresCopyBinaryFieldWriter : public PostgresCopyFieldWriter {
   }
 };
 
+class PostgresCopyBinaryDictFieldWriter : public PostgresCopyFieldWriter {
```

Comment:
It seems like eventually, we'd want to be able to parametrize this writer on the sub-writer, but we can defer that for now